### PR TITLE
Patching sample

### PR DESCRIPTION
**Pull request by jakejscott**
## What was changed
Added sample for Patching

## Why?
Because there was an open issue for it :)

## Checklist

1. Closes https://github.com/temporalio/samples-dotnet/issues/66

2. How was this tested:
Manually copy pasting each command from the samples README

3. Any docs updates needed?
I've updated the repo README.

### CODE REVIEW

File: src/Patching/MyWorkflow1Initial.workflow.cs [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    public async Task RunAsync()
+    {
+        result = await Workflow.ExecuteActivityAsync(() => Activities.PrePatchActivity(), new() { ScheduleToCloseTimeout = TimeSpan.FromMinutes(5) });
+    }
+
+    [WorkflowQuery]
+    public string? Result()
+    {
+        return result;
+    }
```
Comment (cretz) [inline src/Patching/MyWorkflow1Initial.workflow.cs:20]:
```suggestion
    [WorkflowQuery]
    public string? Result { get; private set; }
```

Cleaner to use properties IMO (and can remove field)

**Review by cretz — COMMENTED** [review]
This is great, thanks! Only one slight request of turning that `Result` to a property, but then happy to merge as is. It all reads very well.

File: src/Patching/MyWorkflow1Initial.workflow.cs (hunk 1 above)
Comment (jakejscott) [reply src/Patching/MyWorkflow1Initial.workflow.cs]: Done

File: src/Patching/README.md [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+This will output "Query result for ID initial-workflow-id: pre-patch" since it is pre-patch. But if we execute a query
+against the new code:
+
+    dotnet run starter --query-workflow patched-workflow-id
+
+We get "Query result for ID patched-workflow-id: post-patch". This is how old workflow code can take old paths and new
+workflow code can take new paths.
+
+### Stage 3 - Deprecation
```
[the remark this answers is not in the document]
Comment (jakejscott) [reply src/Patching/README.md:47]: Done

**Review by cretz — APPROVED** [review]
This is great, thanks!

**Comment by cretz** [comment]
Oh, it looks like there is a conflict with the `sln` file. Can you merge `main`, use `main`'s `sln` file, then re `dotnet sln add` this project to fix the conflict?

**Comment by cretz** [comment]
Thanks!

**Pull request merged**